artifactory/commands/utils/transferconfigbase_test.go

```diff
@@ -146,41 +146,88 @@ func TestGetSelectedRepositories(t *testing.T) {
-	assert.NoError(t, err)
+	federatedRepo := readRepoConfig(t, "federated_repo")
+	federatedRepoWithoutMembers := readRepoConfig(t, "federated_repo_without_members")
 
 	testServer, serverDetails, _ := commonTests.CreateRtRestsMockServer(t, func(w http.ResponseWriter, r *http.Request) {
 		w.WriteHeader(http.StatusOK)
-		if r.Method == http.MethodGet {
-			if r.RequestURI == "/api/repositories/federated-local" {
+		switch r.Method {
+		case http.MethodGet:
+			switch r.RequestURI {
+			case "/api/repositories/federated-local":
 				_, err := w.Write(federatedRepo)
 				assert.NoError(t, err)
-			} else if r.RequestURI == "/api/repositories/federated-local-no-members" {
+			case "/api/repositories/federated-local-no-members":
 				_, err := w.Write(federatedRepoWithoutMembers)
 				assert.NoError(t, err)
+			default:
+				assert.Fail(t, "Unexpected request URI "+r.RequestURI)
 			}
-		} else if r.Method == http.MethodPost {
+		case http.MethodPut:
 			body, err := io.ReadAll(r.Body)
 			assert.NoError(t, err)
-
-			var repoParams interface{}
-			assert.NoError(t, json.Unmarshal(body, &repoParams))
-			repoParamsMap, err := InterfaceToMap(repoParams)
-			assert.NoError(t, err)
-			assert.Equal(t, getExpectedFederatedRepoParamsMap(t, federatedRepoWithoutMembers), repoParamsMap)
+			assert.Equal(t, getRepoParamsMap(t, federatedRepoWithoutMembers), getRepoParamsMap(t, body))
+		default:
+			assert.Fail(t, "Unexpected method "+r.Method)
 		}
 	})
 	defer testServer.Close()
 
 	transferConfigBase := createTransferConfigBase(t, serverDetails, serverDetails)
 	assert.False(t, transferConfigBase.FederatedMembersRemoved)
-	err = transferConfigBase.transferSpecificRepositoriesToTarget([]string{"federated-local", "federated-local-no-members"}, utils.Federated)
+	err := transferConfigBase.transferSpecificRepositoriesToTarget([]string{"federated-local", "federated-local-no-members"}, utils.Federated)
 	assert.NoError(t, err)
 	assert.True(t, transferConfigBase.FederatedMembersRemoved)
 }
 
+func TestTransferVirtualRepositoriesToTarget(t *testing.T) {
+	virtualRepoA := readRepoConfig(t, "virtual_repo_a")
+	virtualRepoB := readRepoConfig(t, "virtual_repo_b")
+
+	testServer, serverDetails, _ := commonTests.CreateRtRestsMockServer(t, func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(http.StatusOK)
+		switch r.Method {
+		case http.MethodGet:
+			switch r.RequestURI {
+			case "/api/repositories/a-virtual":
+				_, err := w.Write(virtualRepoA)
+				assert.NoError(t, err)
+			case "/api/repositories/b-virtual":
+				_, err := w.Write(virtualRepoB)
+				assert.NoError(t, err)
+			default:
+				assert.Fail(t, "Unexpected request URI "+r.RequestURI)
+			}
+		case http.MethodPut:
+			fallthrough
+		case http.MethodPost:
+			body, err := io.ReadAll(r.Body)
+			assert.NoError(t, err)
+
+			virtualRepoAParamsMap := getRepoParamsMap(t, virtualRepoA)
```

Comment:
I suggest renaming these variables to clarify these are the original params, and perhaps adding a comment explaining the deletion in case of PUT.